gensim/models/ensemblelda.py

```diff
@@ -0,0 +1,1268 @@
+            if self.memory_friendly_ttda:
+                # remember that this code is inside the worker processes memory,
+                # so self.ttda is the ttda of only a chunk of models
+                pipe.send(self.ttda)
+            else:
+                pipe.send(self.tms)
+
+            pipe.close()
+
+    def asymmetric_distance_matrix_worker(self, worker_id, ttdas_sent, n_ttdas, pipe, threshold, method):
+        """ worker, that computes the distance to all other nodes
+        from a chunk of nodes. https://stackoverflow.com/a/1743350
+        """
+
+        logger.info("Spawned worker to generate {} rows of the asymmetric distance matrix".format(n_ttdas))
+        # the chunk of ttda that's going to be calculated:
+        ttda1 = self.ttda[ttdas_sent:ttdas_sent + n_ttdas]
+        distance_chunk = self.calculate_asymmetric_distance_matrix_chunk(ttda1=ttda1, ttda2=self.ttda,
+                                                                         threshold=threshold,
+                                                                         start_index=ttdas_sent,
+                                                                         method=method)
+        pipe.send((worker_id, distance_chunk))  # remember that this code is inside the workers memory
+        pipe.close()
+
+    def generate_asymmetric_distance_matrix(self, threshold=None, workers=1, method="mass"):
+        """Makes the pairwise distance matrix for all the ttdas
+        from the ensemble.
+
+        Returns the asymmetric pairwise distance matrix that
+        is used in the DBSCAN clustering.
+
+        Afterwards, the model needs to be reclustered for
+        this generated matrix to take effect.
+
+        Parameters
+        ----------
+        threshold : float, optional
+            if threshold is None and method == "rank":
+                threshold = 0.11
+            if threshold is None and method == "mass":
+                threshold = 0.95
+
+            threshold keeps by default 95% of the largest terms by
+            mass. Except the "fast" parameter is "rank", then it
+            just selects that many of the largest terms.
+        workers : number, optional
+            when workers is None, it defaults to os.cpu_count()
+            for maximum performance. Default is 1, which is not
+            multiprocessed. Set to > 1 to enable multiprocessing.
+
+        """
+
+        # matrix is up to date afterwards
+        self.asymmetric_distance_matrix_outdated = False
+
+        if threshold is None:
+            threshold = {"mass": 0.95, "rank": 0.11}[method]
+
+        logger.info("Generating a {} x {} asymmetric distance matrix...".format(len(self.ttda), len(self.ttda)))
+
+        # singlecore:
+        if workers is not None and workers <= 1:
+            self.asymmetric_distance_matrix = self.calculate_asymmetric_distance_matrix_chunk(ttda1=self.ttda,
```

Comment:
Please fix this indentation as described above.

Reply:
if hanging indents are also a valid option, this is done